lib/analytics/apis/analytics_api.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:firebase_analytics/firebase_analytics.dart';
+import 'package:flex_storefront/analytics/models/analytics_events.dart';
+import 'package:get_it/get_it.dart';
+import 'package:loggy/loggy.dart';
+
+mixin AnalyticsApiLoggy implements LoggyType {
+  @override
+  Loggy<AnalyticsApiLoggy> get loggy =>
+      Loggy<AnalyticsApiLoggy>('AnalyticsApiLoggy');
+}
+
+class AnalyticsApi with AnalyticsApiLoggy {
+  Future<void> track(AnalyticsEvent event) async {
+    loggy.info('Track Analytics event $event');
+    if (event is ViewItemEvent) {
+      await GetIt.instance.get<FirebaseAnalytics>().logViewItem(
```

Comment:
Can we just store a local `_firebaseAnalytics` variable privately in this Api and call `GetIt.instance.get` a single time?

Reply:
My problem with that, is that the order of singleton registration becomes very strict in `init.dart`. Because the underlying singletons are looked up in the GetIt registry at creation time.

It causes issues when singleton A depends on singleton B and singleton B depends on singleton A.

With this implementation, the underlying singleton is looked up at method runtime, after init is done and everything is in the registry.

I do admit though that we loose the vision of "which singletons are used by XXXX"...